File: app/schemas/stock.py
from __future__ import annotations

from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class StockBasicResponse(BaseModel):
    stock_code: str
    market: str
    company_name: str
    company_short_name: str | None = None
    industry: str | None = None
    industry_name: str | None = None
    par_value: Decimal | None = None
    capital_amount: int | None = None
    issued_common_shares: int | None = None
    listing_date: date | None = None


class StockSearchItemResponse(BaseModel):
    stock_code: str
    market: str
    company_name: str
    company_short_name: str | None = None
    industry: str | None = None
    industry_name: str | None = None
    listing_date: date | None = None


class StockSearchListResponse(BaseModel):
    items: list[StockSearchItemResponse]
    count: int
    total_count: int

File: taiwan-stock-frontend/src/types/stock.ts
export type MarketType = "TWSE" | "TPEX";

export type StockSearchItem = {
  stock_code: string;
  market: MarketType;
  company_name: string;
  company_short_name?: string | null;
  industry?: string | null;
  industry_name?: string | null;
  listing_date?: string | null;
};

export type StockSearchResponse = {
  items: StockSearchItem[];
  count: number;
  total_count: number;
};

export type StockSearchParams = {
  q: string;
  market?: MarketType;
  limit?: number;
  offset?: number;
};